dev/bots/test.dart

```diff
@@ -1718,6 +1699,22 @@ Future<void> _runFlutterTest(String workingDirectory, {
   }
 }
 
+/// This will force the next run of the Flutter tool (if it uses the provided environment) to
+/// have asserts enabled, by setting an environment variable and deleting the cache.
+void recompileFlutterToolWithAsserts(Map<String, String> pubEnvironment) {
```

Comment:
will this also cause subsequent runs to erroneously re-use the asserts enabled tool snapshot?

Reply:
yeah as i was refactoring this code (this function is just an extract of code that was already there twice) i was thinking what we really should do is put the env variables into the snapshot name somehow

Reply:
you mean something like `flutter_tool_with_asserts.snapshot`?

Reply:
or all FLUTTER_TOOL_ARGS?

Reply:
we could write them to the `flutter_tools.stamp` file

Reply:
yeah, one of those

do you think we should do that in this PR or should i file a bug for now?

Reply:
I filed an issue, up to you whether you want to update that here: https://github.com/flutter/flutter/issues/94892

Reply:
In the interests of keeping this PR simple I'm going to not fix it here but I'll look to see if I can do it as a separate PR.